app/user-status-card/user-status-card.component.ts

```diff
@@ -21,7 +23,36 @@ export class UserStatusCardComponent {
     @Input()
     public user: DoseAmigosUser;
 
+    constructor(
+        private doseAmigosUserService: DoseAmigosUserService,
+        private events: Events
+    ) {
+    }
+
     @Input()
     public onClickPage: Type;
 
+    @Input()
+    public allowDelete: boolean;
+
+    /**
+     * Deletes an amigo.
+     * @param doseAmigoUser to delete.
+     * @returns {Promise<TResult>}
+     */
+    public remove = (doseAmigoUser: DoseAmigosUser) => {
```

Comment:
It would have been more work and time (so not doing) but ideally we would set the loading mask when the delete button clicked instead of when reloading.  That requires more events and error handling however...